text/0276-named-args.md

```diff
@@ -0,0 +1,128 @@
+- Start Date: 2017-12-10
+- RFC PR: https://github.com/emberjs/rfcs/pull/276
+- Ember Issue: https://github.com/emberjs/ember.js/pull/15968
+
+# Summary
+
+Introduce `{{@foo}}` in as a dedicated syntax for a component's template to
+refer to named arguments passed in by the caller.
+
+For example, given the invocation `{{hello-world name="Godfrey"}}` and this
+component template in `app/templates/components/hello-world.hbs`:
+
+```hbs
+Hello, {{@name}}
+```
+
+Ember will render "Hello, Godfrey".
+
+# Motivation
+
+Currently, the way to access named arguments passed in from the caller is to
+reference `{{name}}` in the template. This works because when Ember creates
+the component instance, it automatically [assigns](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign)
+all named arguments as properties on the component instance.
+
+The first problem with this approach is that the `{{name}}` syntax is highly
+ambigious, as it could be referring to a local variable (block param), a
+helper or a named argument from the caller (which actually works by accessing
+auto-reflected `{{this.name}}`) or a property on the component class (such as
```

Comment:
I had no idea! I want to start using this yesterday.

Reply:
`this.name`? you can

Reply:
Now that I think about it, I vaguely remember that `{{this.name}}` was the *right* way to use properties in pre 1.0 or 1.x. Or something about `{{this.name}}` would not be proxied to the underlying model of the controller and `{{name}}` would be (or vice versa). But that is a side topic, don't want to take over this rfc 😄